weaver/database/mongodb.py

```diff
@@ -42,11 +42,11 @@ class MongoDatabase(DatabaseInterface):
     _stores = None
     type = "mongodb"
 
-    def __init__(self, container):
-        # type: (AnySettingsContainer) -> None
-        super(MongoDatabase, self).__init__(container)
-        self._database = get_mongodb_engine(container)
-        self._settings = get_settings(container)
+    def __init__(self, registry, reset_connection=False):
+        # type: (AnySettingsContainer, bool) -> None
+        super(MongoDatabase, self).__init__(registry)
+        self._database = get_mongodb_engine(registry, reset_connection)
+        self._settings = registry.settings
```

Comment:
use original definition which is safer as it handles more variations of 'settings container'

Reply:
go back to the version that didn't used `reset_connection` as parameter?

Reply:
`reset_connection` is ok,  but the `get_settings` and "container" variants are better instead of `registry.settings`

Reply:
Ok yes, using `self._settings = get_settings(registry)` is better than `self._settings = registry.settings`, you're right.
Note that reason why I changed it to `registry` instead of `container` is that `container` is not used anymore in this context, was simply renamed to `registry`